src/sysext/sysext.c

```diff
@@ -594,18 +524,32 @@ static int merge_subprocess(Hashmap *images, const char *workspace) {
                         assert_not_reached("Unsupported image type");
                 }
 
-                r = validate_version(
-                                p,
-                                img->name,
-                                host_os_release_id,
-                                host_os_release_version_id,
-                                host_os_release_sysext_level);
-                if (r < 0)
-                        return r;
-                if (r == 0) {
-                        n_ignored++;
-                        continue;
-                }
+                /* Insist that extension images do not overwrite the underlying OS release file (it's fine if
+                * they place one in /etc/os-release, i.e. where things don't matter, as they aren't
+                * merged.) */
+                r = chase_symlinks("/usr/lib/os-release", p, CHASE_PREFIX_ROOT, NULL, NULL);
+                if (r < 0) {
+                        if (r != -ENOENT)
+                                return log_error_errno(r, "Failed to determine whether /usr/lib/os-release exists in the extension image: %m");
+                } else
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "Extension image contains /usr/lib/os-release file, which is not allowed (it may carry /etc/os-release), refusing.");
+
+                if (!arg_force) {
```

Comment:
you shift the arg_force thing around now. previously we wouldn't even do the chase_symlinks() thing if specified. Now you suddenly do? i.e. previously you could have extensions with osrelease file, as long as you specified --force, which i think is ok'ish, since it doesn't matter if all extensions have the same one